app/models/theme_field.rb

```diff
@@ -716,6 +735,22 @@ def inline_javascript?(node)
       true
     end
   end
+
+  def migration_filename_is_valid
+    if !name.match?(/\A\d{4}-.+/)
+      self.errors.add(
+        :base,
+        I18n.t("themes.import_error.migrations.invalid_filename", filename: name),
+      )
+      return
+    end
+
+    # the 5 here is the length of the first 4 digits and the dash that follows
+    # them
+    if name.size - 5 > MIGRATION_NAME_PART_MAX_LENGTH
```

Comment:
I think we should also enforce some sort of format for the name here so that not all characters can be used in the file name. Maybe something like `-([a-zA-Z0-9\/])`?

Reply:
Done.